### ISSUE TITLE
bug(no-number-schema-with-int): It only works if you use `import * as z from "zod"`

### ISSUE THREAD

**Issue by manferlo81** [marcalexiei/eslint-plugin-zod-x#108]
### 🧩 Description

... when you use this rule (an other rules from this plugin) and you imported zod in a way different from what the plugin expects (`import * as z from "zod"`), they don't work...

### 🪜 How to reproduce

```typescript
import { number } from 'zod'

const Schema = number().int() // this should trigger a linting error, even if it can't be autofixed
```

### ✅ Expected behavior

It should report an error no matter how you use it...

```typescript
import * as z from 'zod'

const Schema = z.number().int() // Linting error! perfect!
```

... and...

```typescript
import { number } from 'zod'

const Schema = number().int() // This should be a linting error as well
```

... the plugin has to be aware that the `number` function is coming from `zod` and therefore report an error.

Maybe it hasn't been implemented as it would be a bit (or a lot) more involving to check for that use case, but I believe it's worth to be considered.

### 💻 System info

```shell
It doesn't matter...
```

**Comment by marcalexiei** [comment]
Detecting this requires to rewrite all rules detect logic since it involves to: 
- process all zod imports imported values 
- store every imported variables that is a schema (requires to map all schema names)
- retrieve every node relevant to the rule
- execute rule checks
- the implementation must be compatible with current rule behaviour which is processing `z.*`

**Comment by marcalexiei** [comment]
Fix available in [v1.10.0](https://github.com/marcalexiei/eslint-plugin-zod-x/releases/tag/v1.10.0).
